File: Common/Messages/PushNotification/SendPushNotificationRequestModel.cs
using System;
using System.ComponentModel.DataAnnotations;
using IOBootstrap.NET.Common.Attributes;
using IOBootstrap.NET.Common.Enumerations;
using IOBootstrap.NET.Common.Messages.Base;

namespace IOBootstrap.NET.Common.Messages.PushNotification;

public class SendPushNotificationRequestModel : IORequestModel
{
    public DeviceTypes DeviceType { get; set; }
    
    [IOBackofficeRequest]
    public string? NotificationCategory { get; set; }
    
    [IOBackofficeRequest]
    public string? NotificationData { get; set; }

    [Required]
    [MaxLength(512)]
    public string? NotificationMessage { get; set; }

    [MaxLength(128)]
    public string? NotificationTitle { get; set; }
}

File: React.UI/modules/iobootstrap-bo-pushnotifications/src/models/SendPushNotificationRequestModel.ts
import { BaseRequestModel, DeviceTypes } from "iobootstrap-ui-base";

class SendPushNotificationRequestModel extends BaseRequestModel {

	deviceType: DeviceTypes;
	notificationCategory: string | null;
	notificationData: string | null;
	notificationMessage: string;
	notificationTitle: string;

	constructor() {
		super();

		this.deviceType = DeviceTypes.Unkown;
		this.notificationCategory = null;
		this.notificationData = null;
		this.notificationMessage = "";
		this.notificationTitle = "";
	}
}

export default SendPushNotificationRequestModel;
